specification/src/specification/schema/scalar-types.md

```diff
@@ -37,6 +36,26 @@ For example:
 }
 ```
 
+### Standard Comparison Operators
+
+#### `Equal`
+
+An operator defined using type `equal` tests if a column value is equal to a scalar value, another column value, or a variable.
+
+The precise semantics of this equality operator are not specified, but the operator should define an equivalence relation. That is, its argument type should be the same scalar type for which it is defined, and at the level of values, it should be reflexive, symmetric and transitive.
```

Comment:
How about specifying some more precise semantics? So that in the engine we can rely on using the equality operator from either connector in case of remote joins?

Reply:
Maybe we can list out all the use-cases where we expect this to be used by the engine? And that implicitly defines the semantics?

Reply:
> How about specifying some more precise semantics? So that in the engine we can rely on using the equality operator from either connector in case of remote joins?

The problem is that it's very difficult in general. You might have two completely unrelated types which happen to serialize to the same JSON representation, like `citext` and `varchar`, but with different equalities. So we can't just move the evaluation from one connector to the other. For local relationships, we can check that the types match, but we can't for remote relationships.

Reply:
As discussed, we will leave this as-is, and handle this on the engine side, or with a breaking change here, when we come to implement more advanced remote joins.